### PR TITLE
MODLD-343: Fix NPE in DataField043Preprocessor

### PR DESCRIPTION
**Pull request by askhat-abishev**

### CODE REVIEW

**Review by PBobylev — APPROVED** [review]

File: src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java [hunk 1]
```diff
@@ -38,6 +39,8 @@ public String getTag() {
 
   @Override
   public boolean isValid(DataField dataField) {
-    return dictionaryProcessor.getValue("NAME", dataField.getSubfield(CODE_A).getData()).isPresent();
+    return Optional.ofNullable(dataField.getSubfield(CODE_A))
```
Comment (aleksei-pronichev) [inline src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java:42]:
```java
return Optional.ofNullable(dataField.getSubfield(CODE_A))
      .map(subfield -> subfield.getData())
      .flatMap(data-> dictionaryProcessor.getValue("NAME", data))
      .isPresent()
```

File: src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java [hunk 2]
```diff
@@ -38,6 +39,8 @@ public String getTag() {
...
+      .map(subfield -> dictionaryProcessor.getValue("NAME", subfield.getData()).isPresent())
```
Comment (aleksei-pronichev) [inline src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java:43]: minor
`NAME` - to constants (like CODE_VALUE) to understand the meaning

File: src/test/resources/full_marc.jsonl [hunk 3]
```diff
@@ -102,6 +102,15 @@
...
     },
+    {
+      "043": {
+        "subfields": [
+          {
+            "k": "asd"
+          }
+        ]
+      }
+    },
```
Comment (aleksei-pronichev) [inline src/test/resources/full_marc.jsonl:113]: minor
I suggest adding separate test cases to improve our test base and general quality.

File: src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java (hunk 1 above)
Comment (askhat-abishev) [reply src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java:45]: done

File: src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java (hunk 2 above)
Comment (askhat-abishev) [reply src/main/java/org/folio/marc4ld/service/marc2ld/preprocessor/impl/DataField043Preprocessor.java]: done

File: src/test/resources/full_marc.jsonl (hunk 3 above)
Comment (askhat-abishev) [reply src/test/resources/full_marc.jsonl:113]: done

**Review by pkjacob — APPROVED** [review]

**Pull request merged**